internal/flow/promote.go

```diff
@@ -256,3 +244,81 @@ func (s *Service) ExecPromote(ctx context.Context, p PromoteEvent) error {
 	}
 	return nil
 }
+
+func (s *Service) releasePaths(ctx context.Context, service, environment, artifactID string) (string, string, func(context.Context), error) {
+	logger := log.WithContext(ctx)
+	sourceConfigRepoPath, closeSource, err := git.TempDirAsync(ctx, s.Tracer, "k8s-config-release-paths")
+	if err != nil {
+		return "", "", nil, err
+	}
+	logger.Debugf("Cloning source config repo %s into %s", s.Git.ConfigRepoURL, sourceConfigRepoPath)
+	sourceRepo, err := s.Git.Clone(ctx, sourceConfigRepoPath)
+	if err != nil {
+		closeSource(ctx)
+		return "", "", nil, errors.WithMessagef(err, "clone into '%s'", sourceConfigRepoPath)
+	}
+	hash, err := s.Git.LocateRelease(ctx, sourceRepo, artifactID)
+	if err != nil {
+		closeSource(ctx)
+		return "", "", nil, errors.WithMessage(err, "locate artifact")
+	}
+	err = s.Git.Checkout(ctx, sourceConfigRepoPath, hash)
+	if err != nil {
+		closeSource(ctx)
+		return "", "", nil, errors.WithMessagef(err, "checkout release hash '%s'", hash)
+	}
+	artifactPath := srcPath(sourceConfigRepoPath, service, "master", s.ArtifactFileName)
+	resourcesPath := srcPath(sourceConfigRepoPath, service, "master", environment)
+	return artifactPath, resourcesPath, closeSource, nil
+}
+
+func (s *Service) releaseExists(ctx context.Context, artifactID string) (bool, error) {
```

Comment:
What does this do? 🤔

Reply:
Check for the existence of a release of provided  artifact ID.